lib/maths/CBoostedTreeImpl.cc

```diff
@@ -362,12 +364,15 @@ std::size_t CBoostedTreeImpl::estimateMemoryUsage(std::size_t numberRows,
 }
 
 std::size_t CBoostedTreeImpl::correctedMemoryUsage(double memoryUsageBytes) {
-    // We compute the correction coefficient as a sigmoid function: ca. 1.0 until
-    // 10mb, ca 16.0 after 1000mb to this end we need to shift and scale using the
-    // magic numbers below.
-    double correctionCoefficient{
-        CTools::logisticFunction(memoryUsageBytes / (1024 * 1024), 100, 550) * 15 + 1};
-    return static_cast<std::size_t>(memoryUsageBytes / correctionCoefficient);
+    // We use a piecewise linear function of the estimated memory usage to compute
+    // the corrected value. The values are selected in a way to reduce over-estimation
+    // on small jobs to improve the behaviour on the trial nodes in the cloud.
+    TDoubleVec estimatedMemoryUsageMB{0., 1024., 4096., 8192., 12288., 16384.};
+    TDoubleVec correctedMemoryUsageMB{0., 179.2, 512., 819.2, 1088., 1280.};
```

Comment:
On reflection, I think it is worth having this corrected directly proportional to the estimate for small data sets. I think this will also fix the failing unit tests. How about: 
```suggestion
    TDoubleVec estimatedMemoryUsageMB{0., 20., 1024., 4096., 8192., 12288., 16384.};
    TDoubleVec correctedMemoryUsageMB{0., 20., 179.2, 512., 819.2, 1088., 1280.};
```

Reply:
Fixed.